WooCommerce/Classes/ViewModels/Order Details/OrderDetailsViewModel.swift

```diff
@@ -409,6 +409,19 @@ extension OrderDetailsViewModel {
         ServiceLocator.stores.dispatch(action)
     }
 
+    func checkCardPaymentEligibility(onCompletion: (() -> Void)? = nil) {
+        // For now, this defaults to the feature flag,
+        // combined with the order payment method being cash on delivery
+        // We also need to take into account wheter the store is enrolled into WCPay.
+        // This is synchronous for now, but it will be async in the nearly future:
+        // https://github.com/woocommerce/woocommerce-ios/issues/3828
+        // We need to add a unit test for this, after the logic is settled
+        dataSource.isEligibleForCardPresentPayment = ServiceLocator.featureFlagService.isFeatureFlagEnabled(.cardPresentPayments) &&
+            OrderPaymentMethod(rawValue: order.paymentMethodID) == .cod
```

Comment:
Same comment above about allowing card present payments when paymentMethod is empty